lntest/itest/signer_test.go

```diff
@@ -0,0 +1,196 @@
+		ctxb, net.Alice, customizedKeyFamily, customizedIndex,
+	)
+	require.NoError(t.t, err, "failed to create customized pubkey")
+
+	// Test DeriveSharedKey with no optional arguments. It will result in
+	// performing an ECDH between the ephemeral key and the node's pubkey.
+	req := &signrpc.SharedKeyRequest{EphemeralPubkey: ephemeralPubBytes}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc which points to the node's pubkey.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyLoc: &signrpc.KeyLocator{
+			KeyFamily: int32(keychain.KeyFamilyNodeKey),
+			KeyIndex:  0,
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc being set in KeyDesc. The KeyLoc
+	// points to the node's pubkey.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: int32(keychain.KeyFamilyNodeKey),
+				KeyIndex:  0,
+			},
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with RawKeyBytes set in KeyDesc. The RawKeyBytes
+	// is the node's pubkey bytes. Because the KeyLoc is nil, it will end up
+	// using the node's pubkey to perform the ECDH.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			RawKeyBytes: net.Alice.PubKey[:],
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc which points to the customized
+	// public key.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyLoc: &signrpc.KeyLocator{
+			KeyFamily: customizedKeyFamily,
+			KeyIndex:  customizedIndex,
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Test DeriveSharedKey with a KeyLoc being set in KeyDesc. The KeyLoc
+	// points to the customized public key.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: customizedKeyFamily,
+				KeyIndex:  customizedIndex,
+			},
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Test DeriveSharedKey with RawKeyBytes set in KeyDesc. The RawKeyBytes
+	// is the customized public key. The KeyLoc is also set with the family
+	// being the customizedKeyFamily.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			RawKeyBytes: customizedPub.SerializeCompressed(),
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: customizedKeyFamily,
+			},
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Now we test DeriveSharedKey with invalid params.
+
+	// assertErrorMatch checks the given error is matched from the response.
+	assertErrorMatch := func(match string, req *signrpc.SharedKeyRequest) {
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		resp, err := net.Alice.SignerClient.DeriveSharedKey(ctxt, req)
+		require.Nil(t.t, resp, "expect no response")
```

Comment:
Should use `require.Error` instead. Then we can be certain that the below `err.Error()` never runs into a nil reference error.

Reply:
fixed.